pkg/cmd/server/origin/run_components.go

```diff
@@ -508,3 +510,31 @@ func (c *MasterConfig) RunClusterQuotaMappingController() {
 		go c.ClusterQuotaMappingController.Run(5, utilwait.NeverStop)
 	})
 }
+
+func (c *MasterConfig) RunClusterQuotaReconciliationController() {
+	osClient, kClient := c.ResourceQuotaManagerClients()
+	resourceQuotaRegistry := quotainstall.NewRegistry(kClient)
+	groupKindsToReplenish := []unversioned.GroupKind{
+		kapi.Kind("Pod"),
+		kapi.Kind("Service"),
+		kapi.Kind("ReplicationController"),
+		kapi.Kind("PersistentVolumeClaim"),
+		kapi.Kind("Secret"),
+		kapi.Kind("ConfigMap"),
```

Comment:
> I will add a utility method upstream to pull this list so we can pull from a common set?

It's more structurally flawed.  The choice should be based on GroupResource and that needs to built based on an intersection with enabled resources.  It doesn't actually have to be limited to this server's hosting, since it could observe changes in federated servers.

In addition, we need to extend our InformerFactory to transparently handle generic listener requests to properly feed this beast.

We should probably restructure the replenishment controller to work as a listener/notifier and upstream the bucketter I built here to be able to handle partial updates.

I haven't prioritized the list of TODOs, but its a significant endeavor.